AsyncDisplayKit/ASControlNode.m

```diff
@@ -339,7 +339,7 @@ - (void)sendActionsForControlEvents:(ASControlNodeEvent)controlEvents withEvent:
   _ASEnumerateControlEventsIncludedInMaskWithBlock(controlEvents, ^
     (ASControlNodeEvent controlEvent)
     {
-      NSMapTable *eventDispatchTable = [_controlEventDispatchTable objectForKey:_ASControlNodeEventKeyForControlEvent(controlEvent)];
+      NSMapTable *eventDispatchTable = [[_controlEventDispatchTable objectForKey:_ASControlNodeEventKeyForControlEvent(controlEvent)] copy];
```

Comment:
Could you add a comment above this, in the style of the file explaining the rationale?  It's not immediately obvious that the action perform would trigger a call into remove, but of course makes sense with a few words of description.

I'm more worried about any performance implication of this, though; copy is probably faster than adding items for the first time, but I've seen the latter prominently in profiling the display system's usage of NSMapTable.

It's just a shame that probably ~10,000+ events will be delivered for 1 that causes an action removal, since most apps won't do this.  It could make sense to set a boolean flag before iterating, and if that flag is encountered at the mutation point, perform the copy and overwrite the local variable with the copy (as well as clear the flag).  I'll land this PR with the project file revert and comment though.